mantidproject/mantid
# Diffraction Reduction - OSIRIS - Remove validation involving comparison of the number of runs

Currently, in the OSIRISDiffractionReduction algorithm, the number of input sample, container and vanadium runs are compared during the validation step (must be more sample than vanadium runs, number of container runs and sample runs must be equal if there are one or more container runs). 

Since runs are no longer combined by the order they are supplied and instead by their respective D-Ranges, thus the smallest intersection of the three sets of runs is used. This validation should be removed, as it is both unnecessary and is producing an error when user's wish to change the number of sample/container/vanadium files and not change the other files (e.g. entering D1-5 for Vanadium and Container and varying the sample runs used without being required to change the other two inputs).

**To test:**
1. Navigate to Interfaces > Indirect > Diffraction.
2. Select 'OSIRIS' in the Instrument drop-down menu.
3. Enter `119976, 119977, 119978, 119981` into the 'Sample' field.
4. Enter `71988, 71991, 71990` into the 'Container' field.
5. Enter `119965, 119967` into the 'Vanadium' field.
**Note the runs are provided in varying orders - this is to test that the order is not considered, runs should be organised by d-range**
6. Provide the supplied calibration file into the 'Calibration' field.
7. Click 'Run'.
8. Open the sample logs of the '\*\_dRange' workspace.
9. Ensure the 'D-Ranges' log value is `3.1 - 4.3 Angstrom, 5.2 - 6.2 Angstrom`

**Test files:**
[osiris_041_RES10.zip](https://github.com/mantidproject/mantid/files/1568749/osiris_041_RES10.zip)



Fixes #21357 . 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
